pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -268,6 +275,36 @@ protected void readMoreEntries() {
         }
     }
 
+    /**
+     * Determine if the messages are continuous, if they are not, the messages are discarded and rewind is called.
+     * Why are the messages discontinuous, for example: if there has something wrong when Replicator is processing
+     * messages "[1:1 ~ 3:3]", Replicator discards the unprocessed message. But a new batch messages "[4:1 ~ 6:6]"
+     * is received later, then these messages will be sent.
+     */
+    protected boolean isMessageContinuousAndRewindIfNot(List<Entry> entries) {
```

Comment:
```suggestion
    protected boolean hasMessageSkipped(List<Entry> entries) {
```

Reply:
The reply of this comment is same as https://github.com/apache/pulsar/pull/20128#discussion_r1171219204